### ISSUE TITLE
Fails on const

### ISSUE THREAD

**Issue by cbaggers** [Shirakumo/glsl-toolkit#3]
```
(glsl-toolkit:parse "void foo() { const int OFFSET = 10; }")
```
> The parse rule SHADER did not consume all of the tokens.
> It is likely that it failed somewhere around VOID (position 0)

Without the `const` the parse succeeds

From the wiki
> Global and local variables, as well as input function parameters, can be declared with the const qualifier.
> Input parameters can be defined with const; ... The const qualifier must come before the in qualifier.

**Comment by cbaggers** [comment]
This let's the parse succeed, the `const` isnt mentioned in the resulting tree
```
modified   grammar.lisp
@@ -286,7 +286,7 @@
       (and :precision (v (or :highp :mediump :lowp)) (v type-specifier) :\;))
 
 (define-object variable-declaration
-      (and (v (? type-qualifier)) (v type-specifier)
+      (and (v (* type-qualifier)) (v type-specifier)
            (v variable-initializer) (* (and :\, (v variable-initializer)))
            :\;)
   (destructuring-bind (qualifier specifier &rest initializers) v
```

**Comment by Shinmera** [comment]
This was fixed by 463190f

**Comment by cbaggers** [comment]
thanks
